### PR TITLE
Shivarthu Milestone 1

### PR DESCRIPTION
**Pull request by amiyatulu**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/525 < please fill this in with the PR number of your application.

### CODE REVIEW

File: deliveries/Shivarthu Milestone 1.md
Comment (takahser) [inline deliveries/Shivarthu Milestone 1.md:9]:
```suggestion
* **Application Document:** [Shivarthu](https://github.com/w3f/Grants-Program/blob/master/applications/Shivarthu.md)
```

**Review by takahser — CHANGES_REQUESTED** [review]

File: deliveries/Shivarthu Milestone 1.md
Comment (takahser) [reply deliveries/Shivarthu Milestone 1.md]: just fixing the build here:)

**Comment by takahser** [comment]
@amiyatulu I took a look at your delivery, but I wasn't able to find the instructions to run and test your code.
I added more detailed comments to my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/2af509ce953dcc0f54e2e3da2c78c84d6eb4f0b6/evaluations/shivarthu_1_takahser.md), please have a look and let me know, if you have any questions.

**Comment by amiyatulu** [comment]
@takahser Thank you for reviewing. Docker compose is failing because non creation of .local folder. I have updated the docker-compose.yml file, please use the recent commit. I will soon give you the tutorial and instructions to test the app.

**Comment by takahser** [comment]
> I will soon give you the tutorial and instructions to test the app.

@amiyatulu did you update the tutorial/instructions yet, so they reflects the latest changes? Please let me know when it's ready to test again.

**Comment by amiyatulu** [comment]
Hi, @takahser sorry for the delay.  Please find the docs here: https://github.com/amiyatulu/shivarthu_frontend/blob/main/docs/Tutorial.md 
https://github.com/amiyatulu/shivarthu/blob/main/README.md 

Please use the latest repository of shivarthu. https://github.com/amiyatulu/shivarthu

**Comment by takahser** [comment]
@amiyatulu thanks for your patience and sorry for the wait here - we have currently a rather large backlog and are a bit behind. I took another look, I added a few more comments regarding the deliveries 0b, 0c to my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/3ed435b1d93ea64e33fc5a148fbca1879876a374/evaluations/shivarthu_1_takahser.md) and I accepted 0d. Let me know if you have any questions.

**Comment by amiyatulu** [comment]
@takahser Thank you for the update. 
### Documentation Feedback
`The template pallet that seems to build on [parity's template pallet](https://paritytech.github.io/substrate/master/src/pallet_template/lib.rs.html#1-102) should be renamed to a more meaningful name.`
I am updating the logic of `template` to into a loose coupling pallet in the next version (https://github.com/amiyatulu/shivarthu/tree/main/pallets/schelling-game-shared) so that logic can be shared to all pallets. I am mainly working on substrate (https://github.com/amiyatulu/shivarthu/tree/main/pallets/profile-validation), need some time for frontend. 

`I'm confused about the reason for why a [light version of the grant application doc](https://github.com/amiyatulu/shivarthu/blob/main/docs/Shivarthu.md) was added to the repo which was also mentioned in the delivery doc.`
I have updated some technical details that is not mentioned in the delivery doc, so made a new version. Not sure if I need to have a pull request for new updates.

`The inline comments are currently very sparse.`
Functions names are self-explanatory, the technical details article contains all details needed. I will add more inline comments if you say, but in the schelling-game-shared and profile-validation, as template pallet will not be present in the future.

### Testing Guide Feedback
`There is (still) no test guide.`
What test guide do you need, about how to run the test?

`However, there are a lot of warnings. Mostly, unused imports and variables, that can easily be fixed:`
Sorry for a lot of warnings of unused imports, as it active development phase, I have not removed it, because it may be needed in updates. I will clean it up, if it's required for milestone delivery. 

Waiting for your feedback. Please let me know what's need to be done further.

**Review by takahser — CHANGES_REQUESTED** [review]
@amiyatulu thanks for your reply.

> I am updating the logic of template to into a loose coupling pallet in the next version

Ok, that works for me. Please resolve it until the next delivery then.

> I have updated some technical details that is not mentioned in the delivery doc, so made a new version.

The problem with this approach is, that it's difficult to track, which version is the current one. I'd suggest you integrate these changes into your original grant application doc by [amending them](https://github.com/w3f/Grants-Program/#changes-to-a-grant-after-approval). This should be easy to approve, especially if you just add features and/or technical details.

> I will add more inline comments if you say, but in the schelling-game-shared and profile-validation

Okay. Yes, please add sufficient comments to help others understand your code, especially in case they want to reuse it for their own project(s).

> What test guide do you need, about how to run the test?

Yes, for completeness it should contain test instructions (probably just `cargo test`).
Also, it should cover instructions on how to test the whole scope of your delivery (so mainly, how to smoke test deliveries 1-4).

**Comment by amiyatulu** [comment]
@takahser Thank you for replying. I have updated the docs and removed most unused imports. Please have a look: https://github.com/amiyatulu/shivarthu

**Comment by takahser** [comment]
@amiyatulu thanks for your patience and sorry for the delay. I checked it out again and tried to run the frontend. However, it's currently blank and showing a console error `Uncaught TypeError: Cannot read properties of undefined (reading 'startsWith')`:

image

The network requests look fine:

image

Also, the node is running and I didn't identify any issue when exploring it in neither of the two frontends:

- https://substrate-developer-hub.github.io/substrate-front-end-template/?rpc=ws://localhost:9944
- https://polkadot.js.org/apps/?rpc=ws%3A%2F%2F127.0.0.1%3A9944#/accounts

Could you double-check if it works on your machine?

**Comment by amiyatulu** [comment]
@takahser Hi, frontend is running fine on my computer. Have you tried the latest commit? I have uploaded the frontend on vercel at https://shivarthuapp.vercel.app/

Updated tutorial: https://github.com/amiyatulu/shivarthu_frontend/blob/main/docs/Tutorial.md

**Review by takahser — CHANGES_REQUESTED** [review]
@amiyatulu thanks for the update.
I pulled the latest changes and tried again - same result.

I'm on the following commits:

```bash
~/repos/shivarthu$ git log
commit 0535df8a374ea66b0411c4c23fe8a0f67937e8ea (HEAD -> main, origin/main, origin/HEAD)
Author: Amiya Behera <amiyatulu@gmail.com>
Date:   Fri Sep 23 16:56:32 2022 +0530

    score game
```

```bash
~/repos/shivarthu_frontend$ git log
commit 371fe81f75267bfd64d186e1f9582a52e76dcddb (HEAD -> main, origin/main, origin/HEAD)
Author: Amiya Behera <amiyatulu@gmail.com>
Date:   Thu Sep 29 13:20:57 2022 +0530

    docs
```

**Comment by takahser** [comment]
@amiyatulu sorry for the delay here, I didn't find time to take a look earlier.
I'll take another round today and give you feedback after.

**Comment by takahser** [comment]
@amiyatulu I tried again and it's still not working. I tested your code on 2 machines (macOS and Ubuntu).
For some reason, now the node doesn't even connect on the [other two frontends](https://github.com/w3f/Grant-Milestone-Delivery/pull/535#issuecomment-1260817996) anymore. Maybe we can schedule a google meet and you can walk me through, if you're interested, to accelerate the process, what do you think?

**Comment by amiyatulu** [comment]
@takahser Ok. This is my email id: amiyatulu at gmail.com Please let me know at what time and day you like to schedule a Google meet.

**Comment by takahser** [comment]
@amiyatulu I've sent you some suggestions via email, please check.

**Review by takahser — APPROVED** [review]
@amiyatulu thanks for your time in yesterday's call and your explanations.
In the meanwhile, I was able to run and smoke-test your delivery and I decided to accept your delivery. However, there are a few things that you should improve for M2. I'm quoting directly from [my evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/shivarthu_1_takahser.md):

> As this is the first delivery, I'm now willing to accept it in the current state, but for M2 these issues should definitely be fixed. I'm referring to:
> 
> - more useful inline comments on the pallet code that can be used to generate documentation using [rustdoc](https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html)
> - removal of the application doc that is out of sync with the original
> - renaming of the template pallet to a more meaningful name
> - fixing of warnings during the build & tests

I'm looking forward your M2 delivery that should incorporate these improvements.
Please let me know if you have any questions.

**Pull request merged**
